### PR TITLE
[SPRING] resolved ambiguous beanValidation Email annotation imports; fix #13379

### LINKED ISSUE #13379: [BUG][JAVA] SpringCodegen (<library>spring-boot</library>) and BeanValidation for type: string and format: email does not work on version 6.0.1
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

We have a property with type: string and format: email and we use Bean Validation for spring-boot.

      properties:
        email:
          type: string
          minLength: 1
          maxLength: 6
          format: email

Using such validations:
                                   `<useBeanValidation>true</useBeanValidation>`
                                    `<performBeanValidation>true</performBeanValidation>`

When we call this:
`mvn -Dmaven.wagon.http.ssl.insecure=true -Dmaven.wagon.http.ssl.allowall=true clean install`

We got compile error:
`Email is ambiguous
[ERROR]   both class org.hibernate.validator.constraints.Email in org.hibernate.validator.constraints and class javax.validation.constraints.Email in javax.validation.constraints match`

while the expected output is: builds and runs successfully

A previous issue addressed the same error [[BUG][JAVA] SpringCodegen (<library>spring-boot</library>) and BeanValidation for type: string and format: email does not work #4786](https://github.com/OpenAPITools/openapi-generator/issues/4786)

##### openapi-generator version

6.0.1

##### Steps to reproduce
`mvn -Dmaven.wagon.http.ssl.insecure=true -Dmaven.wagon.http.ssl.allowall=true clean install`

##### Related issues/PRs

https://github.com/OpenAPITools/openapi-generator/issues/4786

##### Suggest a fix

Using version 5.3.1 and 5.4.0 does not cause this error, but 6.0.1 creates this error.
If we set `<performBeanValidation>` from `true` to `false`, the error is gone.
If we comment out the line ` format: email`, the error is gone.
But we do now know what are the potential downsides of the two methods mentioned above

### PR DESCRIPTION
**Pull request by Mintas**
fix https://github.com/OpenAPITools/openapi-generator/issues/13379

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

https://github.com/OpenAPITools/openapi-generator/issues/13379
mention JavaSpring community:
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaSpring/beanValidationCore.mustache [hunk 1]
```diff
@@ -11,8 +11,10 @@ minLength not set, maxLength set
 }}{{#minItems}}{{^maxItems}}@Size(min = {{minItems}}) {{/maxItems}}{{/minItems}}{{!
 @Size: minItems not set && maxItems set
 }}{{^minItems}}{{#maxItems}}@Size(max = {{.}}) {{/maxItems}}{{/minItems}}{{!
-@Email: useBeanValidation set && isEmail set
-}}{{#useBeanValidation}}{{#isEmail}}@Email{{/isEmail}}{{/useBeanValidation}}{{!
+@Email: useBeanValidation
+}}{{#isEmail}}{{#useBeanValidation}}@javax.validation.constraints.Email{{/useBeanValidation}}{{!
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/JavaSpring/beanValidationCore.mustache:15]: what about jakarta? assuming current imports in model.mustache I assume that you can use only `@Email`(without package name) when we use javax/jakarta annotation, while for Hibernate you should use package + annotation

**Review by borsch — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/JavaSpring/beanValidationCore.mustache (hunk 1 above)
Comment (Mintas) [reply modules/openapi-generator/src/main/resources/JavaSpring/beanValidationCore.mustache:15]: gotcha! gonna fix this, yes

**Review by borsch — APPROVED** [review]

**Comment by tofi86** [comment]
Any updates on this topic? I could need the fix in my project soon...

**Comment by Mintas** [comment]
> #13379

not yet =(
but You can try workaround, described on high level here
https://github.com/OpenAPITools/openapi-generator/issues/13379#issuecomment-1435527082

File: modules/openapi-generator/src/main/resources/JavaSpring/beanValidationCore.mustache [hunk 2]
```diff
@@ -11,8 +11,10 @@ minLength not set, maxLength set
 }}{{#minItems}}{{^maxItems}}@Size(min = {{minItems}}) {{/maxItems}}{{/minItems}}{{!
 @Size: minItems not set && maxItems set
 }}{{^minItems}}{{#maxItems}}@Size(max = {{.}}) {{/maxItems}}{{/minItems}}{{!
-@Email: useBeanValidation set && isEmail set
-}}{{#useBeanValidation}}{{#isEmail}}@Email{{/isEmail}}{{/useBeanValidation}}{{!
+@Email: useBeanValidation
+}}{{#isEmail}}{{#useBeanValidation}}@{{#useJakartaEe}}jakarta{{/useJakartaEe}}{{^useJakartaEe}}javax{{/useJakartaEe}}.validation.constraints.Email{{/useBeanValidation}}{{!
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/JavaSpring/beanValidationCore.mustache:15]: just use `@{{javaxPackage}}.validation.constraints.Email`

**Review by borsch — APPROVED** [review]

**Comment by borsch** [comment]
@Mintas could you please have a look at my latest comment? And also please sync with latest master
Once you do this I'm happy to merge this change

**Comment by deepakthankachan** [comment]
Has this fixed been merged and deployed? Still getting the same error on v7.2.0 :(

**Comment by deepakthankachan** [comment]
We had both performBeanValidation and useBeanValidation set to true and that was throwing the error. Set only one of them to true

**Pull request merged**